### DISCUSSION TITLE
MWS to SP-API Migration

### DISCUSSION THREAD

**Discussion by mohit11115** [amzn/selling-partner-api-models#3316] [General]
Hi,

We have an existing private app running on MWS for downloading and printing our orders.

We got this mail from Amazon informing that MWS is being depreciated and we need to start migrating to SP-API. But it's still not clear if that is meant to be done for only public apps or even our private app.

We are just so confused about the whole process too.

We have been able following this guide:

https://developer-docs.amazon.com/sp-api/docs/what-is-the-selling-partner-api

What we know so far:

1. We need to create a hybrid application to continue using MWS along with SP-API.

What we don't understand yet:

1. What is the Tokens API and do we need it?
2. We were able to generate the LWA refresh token, but how do we include it with the RDT token?

Can someone please guide us through this.

Thanks,

**Comment by prakashjnv90** [comment]
hi @mohit11115 , i can do the migration in case of your looking for developers/consultant.

**Comment by mohit11115** [comment]


**Reply by prakashjnv90** [reply]
Hi @mohit11115 , please drop me an email jprakrn@gmail.com. it would be hard to quantify time without knowing the use cases.

**Comment by sungolivia** [comment]
Hey @mohit11115  have you taken a look at the [migration guide](https://developer-docs.amazon.com/sp-api/docs/amazon-mws-to-sp-api-migration-guide)?

There is also [migration hub ](https://developer-docs.amazon.com/sp-api/docs/amazon-mws-to-sp-api-migration-guide)that mentions all the changes, deadlines that you can keep up to date!
